dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java

```diff
@@ -0,0 +1,222 @@
+import org.reflections.scanners.Scanners;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+public class DocumentationGenerator
+{
+	private static final Logger logger = LoggerFactory.getLogger(DocumentationGenerator.class);
+
+	private static final String ENV_VARIABLE_PLACEHOLDER = "${env_variable}";
+	private static final String PROPERTY_NAME_PLACEHOLDER = " ${property_name}";
+
+	public static void main(String[] args)
+	{
+		new DocumentationGenerator().execute(args);
+	}
+
+	public void execute(String[] args)
+	{
+		Arrays.asList(args).forEach(this::generateDocumentation);
+	}
+
+	private void generateDocumentation(String workingPackage)
+	{
+		String filename = "target/Documentation_" + workingPackage + ".md";
+		logger.info("Generating documentation for package {} in file {}", workingPackage, filename);
+
+		Reflections reflections = createReflections(workingPackage);
+
+		List<String> pluginProcessNames = getPluginProcessNames(reflections, workingPackage);
+		List<Field> fields = getFields(reflections);
+
+		writeFields(fields, pluginProcessNames, filename, workingPackage);
+	}
+
+	private Reflections createReflections(String workingPackage)
+	{
+		ConfigurationBuilder configurationBuilder = new ConfigurationBuilder()
+				.setUrls(ClasspathHelper.forPackage(workingPackage))
+				.setScanners(Scanners.FieldsAnnotated, Scanners.SubTypes);
+		return new Reflections(configurationBuilder);
+	}
+
+	private List<String> getPluginProcessNames(Reflections reflections, String workingPackage)
+	{
+		List<Class<? extends ProcessPluginDefinition>> pluginDefinitionClasses = new ArrayList<>(
+				reflections.getSubTypesOf(ProcessPluginDefinition.class));
+
+		if (pluginDefinitionClasses.size() < 1)
+		{
+			logger.warn("No ProcessPluginDefinitions found in package {}", workingPackage);
+			return Collections.emptyList();
+		}
+
+		if (pluginDefinitionClasses.size() > 1)
+			logger.warn("Found {} ProcessPluginDefinitions ({}) in package {}, using only the first",
+					pluginDefinitionClasses.size(), pluginDefinitionClasses, workingPackage);
+
+		try
+		{
+			ProcessPluginDefinition processPluginDefinition = pluginDefinitionClasses.get(0).getConstructor()
+					.newInstance();
+
+			return processPluginDefinition.getBpmnFiles().map(this::getProcessName).filter(Optional::isPresent)
+					.map(Optional::get).collect(toList());
+		}
+		catch (Exception exception)
+		{
+			logger.warn(
+					"Could not read process names from package {} and ProcessPluginDefinition with name {}, reason is '{}'",
+					workingPackage, pluginDefinitionClasses.get(0).getName(), exception.getMessage());
+			return Collections.emptyList();
+		}
+	}
+
+	private Optional<String> getProcessName(String bpmnFile)
+	{
+		try
+		{
+			InputStream resource = getClass().getClassLoader().getResource(bpmnFile).openStream();
```

Comment:
Make sure the InputStream is being closed somewhere. If it does not happen inside Bpmn.readModelFromStream use try-with resources:

```suggestion
		try (InputStream resource = getClass().getClassLoader().getResource(bpmnFile).openStream())
		{
```

Reply:
resolved with 6a6eb6f